bin/src/command_line.dart

```diff
@@ -1,17 +1,102 @@
 library cmdpart;
 
 import 'dart:async';
+import 'dart:io' show Directory, File, FileSystemEntity, Platform;
 import "dart:math" show max;
 
 import "package:args/args.dart";
+import 'package:dart_coveralls/dart_coveralls.dart' show CommandLineClient, log;
+import 'package:logging/logging.dart';
+import 'package:stack_trace/src/chain.dart';
 
 export "package:args/args.dart";
 
+final Logger _log = new Logger("dart_coveralls");
+
 abstract class CommandLinePart {
   final ArgParser parser;
 
   CommandLinePart(this.parser);
 
+  bool handleLogging(ArgResults res) {
+    var logLevelStr = res['log-level'];
+
+    Level logLevel = Level.LEVELS
+        .firstWhere((level) => level.name.toLowerCase() == logLevelStr);
+
+    if (res['debug']) {
+      if (logLevel == Level.OFF) {
+        logLevel = Level.ALL;
+      } else {
+        print("Cannot set both `--log-level` and `--debug`.");
+        return false;
+      }
+    }
+
+    if (logLevel != Level.OFF) {
+      log.onRecord.where((rec) => rec.level >= logLevel).listen((rec) {
+        print(rec.message);
+        if (rec.error != null) {
+          print(rec.error);
+        }
+        if (rec.stackTrace != null) {
+          print(new Chain.forTrace(rec.stackTrace).terse);
+        }
+      });
+    }
+
+    return true;
+  }
+
+  /// Performs checks on options --packages, --package-root and --token
+  /// before returning a `CommandLineClient` instance.
+  CommandLineClient getCommandLineClient(ArgResults res) {
```

Comment:
@eseidel  this is a modification of your approach to include handling of --packages and --package-root, and supplying 'CommandLineClient` with the right arguments.